extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala

```diff
@@ -43,6 +43,8 @@ class RuleApplyPermanentViewMarker extends Rule[LogicalPlan] {
             //  as the main query part also secures it. But for performance consideration,
             //  we also pre-check it in subqueries and fail fast with negative privileges.
             scalarSubquery.copy(plan = PermanentViewMarker(scalarSubquery.plan, null))
+          case listQuery: ListQuery =>
```

Comment:
use SubqueryExpression for both ScalarSubquery and ListQuery?

Reply:
> SubqueryExpression

Then can't use copy method.

Reply:
You can use withNewPlan instead

Reply:
> You can use withNewPlan instead

Cool ! changed.